### PR TITLE
Docs update

### PR DESCRIPTION
**Pull request by josephlewis42**
* Updated the README with new wording, links to files and extracted installation details into a separate file because it was hiding other relevant info.
* Updated the changelog to go in reverse chronological order and updated 4.0 section.
* Updated OSDF and included documentation about how to generate it.
* Migrated to the official Google contributing doc (the old doc was completely missing info about the CLAs that authors had to sign and was for the old PR process).

Tagging issue #240.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,88 +1,89 @@
...
+- **Breaking** The GCS plan `reduced_availability` was changed to `reduced-availability` to be compliant with the spec and work with Kubernetes.
+- Tables are created only if they do not exist on migration, fixing [#194](https://github.com/GoogleCloudPlatform/gcp-service-broker/issues/194).
+- The broker now adheres to OSB version 2.13.
 
-## [2.0.2] - 2016-11-16
+## [3.6.0] - 2017-01-03
 
-### Fixed
-- fixed bug where CloudSQL was returning 400s for all 2nd gen instance provision requests
+- changed default authorized networks for cloudsql instances from `0.0.0.0/0` to none
```
Comment (craigatgoogle) [inline CHANGELOG.md]: Nit, the use of capitalization and periods in this section deviates from the rest of the doc.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,88 +1,89 @@
...
+- added optional parameter to cloudsql provision operation to specify authorized networks
```
Comment (craigatgoogle) [inline CHANGELOG.md]: s/cloudsql/CloudSQL for consistency

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,88 +1,89 @@
...
+- added service account and key provisioning on cloudsql bind operations
+- updated postgres `uri` field to include ssl certificates
```
Comment (craigatgoogle) [inline CHANGELOG.md]: s/ssl/SSL

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,88 +1,89 @@
...
+- added optional parameter to cloudsql bind operation to pass back jdbc formatted `uri` field
```
Comment (craigatgoogle) [inline CHANGELOG.md]: s/cloudsql/CloudSQL
s/jdbc/JDBC

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,88 +1,89 @@
...
+- removed waiting for ssl certs to finish being created in sql account manager
```
Comment (craigatgoogle) [inline CHANGELOG.md]: s/ssl/SSL
s/sql/SQL

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,88 +1,89 @@
...
+- changed default number of spanner nodes to 1 in `tile.yml`
 
-## [2.1.0] - 2016-12-02
+## [3.5.2] - 2017-10-17
 
-- Remove need for service name for PubSub (topic_name), BigQuery (name), Cloud Storage (name), and Cloud SQL (instance_name)
-- Instance details are now surfaced in bind requests for Pubsub (topic_name and subscription_name),
-Cloud Storage (bucket_name), BigQuery (dataset_id), and Cloud SQL (instance_name, database_name, and host)
+- fixed Postgres connection uri
```
Comment (craigatgoogle) [inline CHANGELOG.md]: s/Postgres/PostgreSQL

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,88 +1,89 @@
...
+- added wait for ssl certs to finish being created in sql account manager
```
Comment (craigatgoogle) [inline CHANGELOG.md]: s/ssl/SSL
s/sql/SQL

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,88 +1,89 @@
...
 
-- fixed a bug in reading in custom plans
+- Added Stackdriver Debugger and Trace support
 
-## [3.1.2] - 2017-02-12
+## [3.2.1] - 2017-03-31
 
-- fixed a bug where supplying a custom name for Bigtable would cause an error
+- fixed a bug where Spanner instances could be deleted from Google but not deprovisioned in CF
+- fixed a bug where Cloud SQL instances were not being marked as deleted in the Service Broker database
```
Comment (craigatgoogle) [inline CHANGELOG.md]: s/Cloud SQL/CloudSQL

File: CHANGELOG.md [hunk 9]
```diff
@@ -91,66 +92,81 @@ any extra access details
...
+### Fixed
+- fixes a bug where anything that triggered an install repeat after installing version 2.1.0 or 2.1.1 would cause the
+installation to fail.
 
-## [3.5.1] - 2017-09-06
+## [2.1.1] - 2016-12-02
 
-- added PostgreSQL support to CloudSQL (and migrated existing plans)
-- added Datastore support
+- added new uri parameter to Cloud SQL bind credentials
```
Comment (craigatgoogle) [inline CHANGELOG.md]: s/Cloud SQL/CloudSQL

File: CHANGELOG.md [hunk 10]
```diff
@@ -91,66 +92,81 @@ any extra access details
...
 
+## [2.1.0] - 2016-12-02
 
-## [3.5.2] - 2017-10-17
+- Remove need for service name for PubSub (topic_name), BigQuery (name), Cloud Storage (name), and Cloud SQL (instance_name)
```
Comment (craigatgoogle) [inline CHANGELOG.md]: s/Cloud SQL/CloudSQL

File: CHANGELOG.md [hunk 11]
```diff
@@ -91,66 +92,81 @@ any extra access details
...
+- Instance details are now surfaced in bind requests for Pubsub (topic_name and subscription_name),
+Cloud Storage (bucket_name), BigQuery (dataset_id), and Cloud SQL (instance_name, database_name, and host)
```
Comment (craigatgoogle) [inline CHANGELOG.md]: s/Cloud SQL/CloudSQL

File: CHANGELOG.md [hunk 12]
```diff
@@ -91,66 +92,81 @@ any extra access details
...
 
-- added PostgreSQL support to CloudSQL (and migrated existing plans)
-- added Datastore support
+- added new uri parameter to Cloud SQL bind credentials
 
+## [2.1.0] - 2016-12-02
 
-## [3.5.2] - 2017-10-17
+- Remove need for service name for PubSub (topic_name), BigQuery (name), Cloud Storage (name), and Cloud SQL (instance_name)
+- Instance details are now surfaced in bind requests for Pubsub (topic_name and subscription_name),
```
Comment (craigatgoogle) [inline CHANGELOG.md]: If we want to keep our naming consistent with product documentation: s/Pubsub/"Pub/Sub"

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+
+You generally only need to submit a CLA once, so if you've already submitted one
+(even if it was for a different project), you probably don't need to do it
+again.
+
+## Code reviews
+
+All submissions, including submissions by project members, require review. We
+use GitHub pull requests for this purpose. Consult
+[GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
```
Comment (craigatgoogle) [inline CONTRIBUTING.md]: s/GitHub Help/GitHub PR Help

File: docs/installation.md [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+    * [Set up a backing database](#database)
+    * [Set required env vars](#required-env)
+    * [Optional env vars](#optional-env)
+    * [Push the service broker to CF and enable services](#push)
+    * [(Optional) Increase the default provision/bind timeout](#timeout)
+
+
+### Installing as a Pivotal Ops Manager tile
+
+Documentation for installing as a Pivotal Ops Manager tile is available [here](http://docs.pivotal.io/partners/gcp-sb/index.html)
```
Comment (craigatgoogle) [inline docs/installation.md]: Missing a period.

File: docs/installation.md [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+
+### [Installing as a Cloud Foundry Application](#cf)
+
+#### [Set up a GCP Project](#project)
+
+1. Go to [Google Cloud Console](https://console.cloud.google.com) and sign up, walking through the setup wizard
```
Comment (craigatgoogle) [inline docs/installation.md]: Inconsistent use of periods in this section

File: docs/installation.md [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+1. From the GCP console, navigate to **IAM & Admin > Service accounts** and click **Create Service Account**.
+1. Enter a **Service account name**.
+1. Select the checkbox to **Furnish a new Private Key**, and then click **Create**.
+1. Save the automatically downloaded key file to a secure location.
+1. Navigate to **IAM & Admin > IAM** and locate your service account.
+1. From the dropdown on the right, choose **Project > Owner** and click **Save**.
+
+#### [Set up a backing database](#database)
+
+1. Create new MySQL instance
```
Comment (craigatgoogle) [inline docs/installation.md]: Same as above, I'll stop repeating myself now, you get the gist :)

File: docs/installation.md [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+* `ROOT_SERVICE_ACCOUNT_JSON` (the string version of the credentials file created for the Owner level Service Account)
+* `SECURITY_USER_NAME` (a username to sign all service broker requests with - the same one used in cf create-service-broker)
+* `SECURITY_USER_PASSWORD` (a password to sign all service broker requests with - the same one used in cf create-service-broker)
+* `DB_HOST` (the host for the database to back the service broker)
+* `DB_USERNAME` (the database username for the service broker to use)
+* `DB_PASSWORD` (the database password for the service broker to use)
+
+#### [Optional env vars](#optional-env)
+
+See https://github.com/GoogleCloudPlatform/gcp-service-broker/blob/master/docs/customization.md
```
Comment (craigatgoogle) [inline docs/installation.md]: Does GitHub automatically linkify raw URLs like this?

File: docs/installation.md [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+for instructions on providing database name and port overrides, ssl certs, and custom service plans for Cloud SQL, Bigtable, and Spanner.
```
Comment (craigatgoogle) [inline docs/installation.md]: s/Cloud SQL/CloudSQL

**Comment by josephlewis42** [comment]
@craigatgoogle thanks for taking a look, I think I've fixed most of the issues.

**Pull request merged**